src/containers/CreatePipelineRun/YAMLEditor.test.js

```diff
@@ -0,0 +1,220 @@
+        {
+          name: 'hello',
+          taskSpec: {
+            steps: [
+              {
+                image: 'busybox',
+                name: 'echo',
+                script: '#!/bin/ash\necho "Hello World!"\n'
+              }
+            ]
+          }
+        }
+      ]
+    }
+  }
+};
+
+describe('YAMLEditor', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.spyOn(window.history, 'pushState');
+    // WA for codemirror vs jsdom https://github.com/jsdom/jsdom/issues/3002#issuecomment-1118039915
+    // for textRange(...).getClientRects is not a function
+    Range.prototype.getBoundingClientRect = () => ({
+      bottom: 0,
+      height: 0,
+      left: 0,
+      right: 0,
+      top: 0,
+      width: 0
+    });
+    Range.prototype.getClientRects = () => ({
+      item: () => null,
+      length: 0,
+      [Symbol.iterator]: jest.fn()
+    });
+  });
+  it('handles onClose event', () => {
+    const { getByText } = renderWithRouter(<CreateYAMLEditor />);
+    fireEvent.click(getByText(/cancel/i));
+    // will be called once for render (from test utils) and once for navigation
+    expect(window.history.pushState).toHaveBeenCalledTimes(2);
+  });
+
+  it('handles close', () => {
+    const { getByText } = renderWithRouter(<CreateYAMLEditor />);
+    fireEvent.click(getByText(/cancel/i));
+    expect(window.history.pushState).toHaveBeenCalledTimes(2);
+  });
+
+  it('handle submit pipelinerun with empty pipelinerun', () => {
+    const { queryAllByText, getByRole, getByText } = renderWithRouter(
+      <CreateYAMLEditor />
+    );
+    expect(getByRole(/textbox/)).toBeTruthy();
+
+    expect(submitButton(queryAllByText)).toBeTruthy();
+
+    fireEvent.click(submitButton(queryAllByText));
+    expect(getByText(/Please fix errors, then resubmit/)).toBeTruthy();
+    expect(getByText('PipelineRun cannot be empty')).toBeTruthy();
+  });
+
+  it('handle submit pipelinerun without namespace', () => {
+    const { queryAllByText, getByRole, getByText } = renderWithRouter(
+      <CreateYAMLEditor />
+    );
+
+    fireEvent.paste(getByRole(/textbox/), {
+      target: { textContent: pipelineRunWithoutNamespace }
+    });
+
+    expect(submitButton(queryAllByText)).toBeTruthy();
+
+    fireEvent.click(submitButton(queryAllByText));
+    expect(getByText(/Please fix errors, then resubmit/)).toBeTruthy();
+    expect(getByText('Namespace cannot be empty')).toBeTruthy();
+  });
+
+  it('handle submit pipelinerun incorrect yaml', () => {
+    const { queryAllByText, getByRole, getByText } = renderWithRouter(
+      <CreateYAMLEditor />
+    );
+
+    fireEvent.paste(getByRole(/textbox/), {
+      target: { textContent: pipelineRunIncorrectYaml }
+    });
+
+    expect(submitButton(queryAllByText)).toBeTruthy();
+
+    fireEvent.click(submitButton(queryAllByText));
+    expect(getByText(/Please fix errors, then resubmit/)).toBeTruthy();
+    expect(getByText(/can not read a block mapping entry/)).toBeTruthy();
```

Comment:
Is this message coming from the client's attempt to parse the YAML via js-yaml:load?

Reply:
Yup, one of those
https://github.com/nodeca/js-yaml/blob/ab31bba6b41f58390f431123ffec5031b986edf5/lib/loader.js#L1136